updatecli/scripts/updateGomodGoversion.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+## This script updates the go version of a go mod file and prints the resulting content in the stdout
+## Please note that it will NEVER change the go.mod file (works in a temp directory)
+##
+## Expected arguments:
+## - 1: the path to the go.mod file
+## - 2: the new golang version in semver format (major.minor.patch)
+
+set -eux
+
+go_mod_dir="$(dirname "${1}")"
+go_version="${2}"
+new_version="$(echo "${go_version}" | cut -d. -f1,2)"
+tmp_dir="$(mktemp -d)"
+
+## Ensures that there is a golang version installed
+{
+  if ! command -v go
+  then
+    curl --silent --show-error --location --output "${tmp_dir}/go.tgz" \
+      "https://golang.org/dl/go${go_version}.$(uname -s | tr '[:upper:]' '[:lower:]')-amd64.tar.gz"
+    mkdir -p "${tmp_dir}/.bin"
+    tar xzf "${tmp_dir}/go.tgz" -C "${tmp_dir}/.bin"
+    export PATH="${PATH}":"${tmp_dir}"/.bin/go/bin
+  fi
+
+  command -v go
+  cat /etc/*release*
+  ldd $(which go)
+  go version
+} >&2
```

Comment:
Important note: downloading golang does not seem to work out of the box on the Alpine Linux agent used by the `updatecli()` shared library. @smerle33 is validating this by trying the script on a local `alpine:3.15` docker image but the `ldd` command added (on puprose) to debug this clearly shows that behavior (Musl instead of glibc).

Proposal is to get rid of the "download `go`" section and only rely on the golang version in the agent environment: if the current`go version` is not able to update the go.mod, then there is no reason to bump the go.mod.

However, it might causes a problem: we have to be sure that the agent environment used by the `updatecli()` command has Golang installed. As for today, we have 2 different agent environments:

- The terraform pipelines are using `jenkinsciinfra/hashicorp-tools` (as per https://github.com/jenkins-infra/pipeline-library/blob/master/vars/terraform.groovy#L11) which has golang version installed (and updatecli wink wink)
- While actually, the pipeline library `updatecli()` uses `jenkinsciinfra/helmfile` by default as per https://github.com/jenkins-infra/pipeline-library/blob/master/vars/updatecli.groovy#L6

=> Hopefully, we should be able to specify the `jenkinsciinfra/hashicorp-tools` image to the `updatecli()` library (something like `updatecli(action: 'diff', updatecliDockerImage: 'jenkinsciinfra/hashicorp-tools:<version>')` should suffice + keeping it up to date :) )